# HG changeset patch
# User Jacob Dawid <jacob.dawid@googlemail.com>
# Date 1313667380 -7200
#      Thu Aug 18 13:36:20 2011 +0200
# Node ID e744593197ef314b63133921101118ab9605925c
# Parent  fede1c59705ac7d2c2b19779ca8e2ea4f056c246
added scrollToBottomRequest signal for terminal and flipped around progress bar an dstatus bar in browser widget, so the handle is on the right corner.

diff --git a/gui/src/BrowserWidget.cpp b/gui/src/BrowserWidget.cpp
--- a/gui/src/BrowserWidget.cpp
+++ b/gui/src/BrowserWidget.cpp
@@ -51,18 +51,18 @@ BrowserWidget::construct ()
   m_navigationToolBar->addWidget (m_urlLineEdit);
 
   QVBoxLayout *layout = new QVBoxLayout ();
   layout->addWidget (m_navigationToolBar);
   layout->addWidget (m_webView);
 
     QWidget *bottomWidget = new QWidget (this);
     QHBoxLayout *bottomLineLayout = new QHBoxLayout ();
+    bottomLineLayout->addWidget (m_progressBar);
     bottomLineLayout->addWidget (m_statusBar);
-    bottomLineLayout->addWidget (m_progressBar);
     bottomLineLayout->setMargin (0);
     bottomWidget->setLayout (bottomLineLayout);
 
   layout->addWidget (bottomWidget);
   layout->setMargin (2);
   setLayout (layout);
 
   connect (backAction, SIGNAL (triggered ()), m_webView, SLOT (back ()));
diff --git a/gui/src/terminal/QTerminalWidget.cpp b/gui/src/terminal/QTerminalWidget.cpp
--- a/gui/src/terminal/QTerminalWidget.cpp
+++ b/gui/src/terminal/QTerminalWidget.cpp
@@ -76,16 +76,19 @@ QTerminalWidget::QTerminalWidget (int st
   if (startnow && m_impl->m_session)
     {
       m_impl->m_session->run ();
     }
 
   setFocus (Qt::OtherFocusReason);
   m_impl->m_terminalDisplay->resize (this->size ());
   setFocusProxy (m_impl->m_terminalDisplay);
+
+  connect (m_impl->m_terminalDisplay, SIGNAL (keyPressedSignal (QKeyEvent*)),
+           this, SLOT (terminalKeyPressed (QKeyEvent*)));
 }
 
 void
 QTerminalWidget::initialize ()
 {
   m_impl->m_terminalDisplay->setSize (80, 40);
 
   QFont font = QApplication::font ();
@@ -153,8 +156,15 @@ QTerminalWidget::resizeEvent (QResizeEve
   m_impl->m_terminalDisplay->update ();
 }
 
 void
 QTerminalWidget::sessionFinished ()
 {
   emit finished ();
 }
+
+void
+QTerminalWidget::terminalKeyPressed (QKeyEvent *keyEvent)
+{
+  Q_UNUSED (keyEvent);
+  emit scrollToBottomRequest ();
+}
diff --git a/gui/src/terminal/QTerminalWidget.h b/gui/src/terminal/QTerminalWidget.h
--- a/gui/src/terminal/QTerminalWidget.h
+++ b/gui/src/terminal/QTerminalWidget.h
@@ -1,20 +1,20 @@
 /*  Copyright (C) 2008 e_k (e_k@users.sourceforge.net)
 
     This library is free software; you can redistribute it and/or
     modify it under the terms of the GNU Library General Public
     License as published by the Free Software Foundation; either
     version 2 of the License, or (at your option) any later version.
-		    
+
     This library is distributed in the hope that it will be useful,
     but WITHOUT ANY WARRANTY; without even the implied warranty of
     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
     Library General Public License for more details.
-			    
+
     You should have received a copy of the GNU Library General Public License
     along with this library; see the file COPYING.LIB.  If not, write to
     the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
     Boston, MA 02110-1301, USA.
 */
 
 #ifndef QTERMINALWIDGET_H
 #define QTERMINALWIDGET_H
@@ -23,43 +23,47 @@
 
 struct TermWidgetImpl;
 /**
   * \class QTerminalWidget
   * This class forms a widget class that can be inserted into other widgets.
   */
 class QTerminalWidget:public QWidget
 {
-Q_OBJECT
-  public:
-    QTerminalWidget (int startnow = 1, QWidget * parent = 0);
-   ~QTerminalWidget ();
+  Q_OBJECT
+public:
+  QTerminalWidget (int startnow = 1, QWidget * parent = 0);
+  ~QTerminalWidget ();
 
   void openTeletype (int fd);
 
-    /** Text codec, default is UTF-8. */
+  /** Text codec, default is UTF-8. */
   void setTextCodec (QTextCodec * codec);
 
-    /** Resize terminal widget. */
+  /** Resize terminal widget. */
   void setSize (int h, int v);
 
-    /** History size for scrolling, values below zero mean infinite. */
+  /** History size for scrolling, values below zero mean infinite. */
   void setHistorySize (int lines);
 
-    /** Send some text to the terminal. */
+  /** Send some text to the terminal. */
   void sendText (const QString & text);
 
-    signals:
-    /** Emitted, when the current program has finished. */
+signals:
+  /** Emitted, when the current program has finished. */
   void finished ();
 
+  void scrollToBottomRequest ();
+
 protected:
-    virtual void resizeEvent (QResizeEvent *);
+  virtual void resizeEvent (QResizeEvent *);
 
-  protected slots:void sessionFinished ();
+protected slots:
+  void sessionFinished ();
+  void terminalKeyPressed (QKeyEvent *keyEvent);
 
 private:
-    /** Performs initial operations on this widget. */
+  /** Performs initial operations on this widget. */
   void initialize ();
   TermWidgetImpl *m_impl;
 };
 
 #endif // QTERMINALWIDGET_H
